src/methods/indexed-db.js

```diff
@@ -99,16 +99,19 @@ export function getAllMessages(db) {
     });
 }
 
-export function getMessagesHigherThen(db, lastCursorId) {
+export function getMessagesHigherThan(db, lastCursorId) {
     const objectStore = db.transaction(OBJECT_STORE_ID).objectStore(OBJECT_STORE_ID);
     const ret = [];
-    const keyRangeValue = IDBKeyRange.bound(lastCursorId + 1, Infinity);
     return new Promise(res => {
-        objectStore.openCursor(keyRangeValue).onsuccess = ev => {
+        objectStore.openCursor().onsuccess = ev => {
             const cursor = ev.target.result;
             if (cursor) {
-                ret.push(cursor.value);
-                cursor.continue();
+                if (cursor.value.id < lastCursorId + 1) {
+                    cursor.continue(lastCursorId + 1);
+                } else {
+                    ret.push(cursor.value);
+                    cursor.continue();
+                }
```

Comment:
This is the change. It opens a cursor on the store without a KeyRange, then advances the cursor. This is because creating the key range can sometimes error in Safari.